test/zdtm.py

```diff
@@ -2566,179 +2561,196 @@ def clean_stuff(opts):
+                    action='store_true')
+    rp.add_argument("--script", help="Add script to get notified by criu")
+    rp.add_argument("-k",
+                    "--keep-img",
+                    help="Whether or not to keep images after test",
+                    choices=['always', 'never', 'failed'],
+                    default='failed')
+    rp.add_argument("--report", help="Generate summary report in directory")
+    rp.add_argument("--keep-going",
+                    help="Keep running tests in spite of failures",
+                    action='store_true')
+    rp.add_argument("--ignore-taint",
+                    help="Don't care about a non-zero kernel taint flag",
+                    action='store_true')
+    rp.add_argument("--lazy-pages",
+                    help="restore pages on demand",
+                    action='store_true')
+    rp.add_argument("--lazy-migrate",
+                    help="restore pages on demand",
+                    action='store_true')
+    rp.add_argument("--remote-lazy-pages",
+                    help="simulate lazy migration",
+                    action='store_true')
+    rp.add_argument("--tls", help="use TLS for migration", action='store_true')
+    rp.add_argument("--title", help="A test suite title", default="criu")
+    rp.add_argument("--show-stats",
+                    help="Show criu statistics",
+                    action='store_true')
+    rp.add_argument("--criu-bin",
+                    help="Path to criu binary",
+                    default='../criu/criu')
+    rp.add_argument("--crit-bin",
+                    help="Path to crit binary",
+                    default='../crit/crit')
+    rp.add_argument("--criu-image-streamer-dir",
+                    help="Directory where the criu-image-streamer binary is located",
+                    default="../../criu-image-streamer")
+    rp.add_argument("--pre-dump-mode",
+                    help="Use splice or read mode of pre-dumping",
+                    choices=['splice', 'read'],
+                    default='splice')
+
+    lp = sp.add_parser("list", help="List tests")
+    lp.set_defaults(action=list_tests)
+    lp.add_argument('-i',
+                    '--info',
+                    help="Show more info about tests",
+                    action='store_true')
+
+    gp = sp.add_parser("group", help="Generate groups")
+    gp.set_defaults(action=group_tests)
+    gp.add_argument("-m", "--max-size",
+                    help="Maximum number of tests in group")
+    gp.add_argument("-n", "--name", help="Common name for group tests")
+    gp.add_argument("-x",
+                    "--exclude",
+                    help="Exclude tests from --all run",
+                    action='append')
+
+    cp = sp.add_parser("clean", help="Clean something")
+    cp.set_defaults(action=clean_stuff)
+    cp.add_argument("what", choices=['nsroot'])
+
+    return vars(p.parse_args())
+
+
+def collect_zombies(pid):
```

Comment:
This is not a good name for this function. The main idea of it is to wait for the target process.

Reply:
@Snorch also mentioned that this may not a good name (in https://github.com/checkpoint-restore/criu/pull/1581#discussion_r731994285). I am happy to change it.

@avagin Do you have any suggestions?

Reply:
Note for @avagin: untill we've waited for a target process (which can run for a long time without exiting) we are also collecting zombies from this function, and we only stop collecting them when target process stops, that's why I changed my mind and said that I'm now ok with the name of the function. Also note that previously this code block which collected zombies in this "partial" maner had a comment proving that it meant to be zombies collection there:
```
-    # Fork here, since we're new pidns init and are supposed to
-    # collect this namespace's zombies
```

More over I believe we should review only last patch in this draft PR, other patches are all from the original https://github.com/checkpoint-restore/criu/pull/1581 and should be reviewed there.

> This pull request is rebased on top of #1581.

Reply:
Don't try to convince me that this is a good name. No, it isn't. The name has to describe what function is for.

> Do you have any suggestions?

waitpid_and_rip_zombies

Reply:
I've renamed the function in https://github.com/checkpoint-restore/criu/pull/1581